src/Microsoft.Identity.Web.Certificateless/AzureIdentityForKubernetesClientAssertion.cs

```diff
@@ -14,40 +15,79 @@ namespace Microsoft.Identity.Web
     /// in Azure Kubernetes Services. See https://aka.ms/ms-id-web/certificateless and
     /// https://learn.microsoft.com/azure/aks/workload-identity-overview
     /// </summary>
-    public class AzureIdentityForKubernetesClientAssertion : ClientAssertionProviderBase
+    public partial class AzureIdentityForKubernetesClientAssertion : ClientAssertionProviderBase
     {
         /// <summary>
         /// Gets a signed assertion from Azure workload identity for kubernetes. The file path is provided
         /// by an environment variable ("AZURE_FEDERATED_TOKEN_FILE")
         /// See https://aka.ms/ms-id-web/certificateless.
         /// </summary>
-        public AzureIdentityForKubernetesClientAssertion() : this(null)
+        public AzureIdentityForKubernetesClientAssertion(ILogger? logger = null) : this(null, logger)
         {
         }
 
         /// <summary>
         /// Gets a signed assertion from a file.
         /// See https://aka.ms/ms-id-web/certificateless.
         /// </summary>
-        /// <param name="filePath"></param>
-        public AzureIdentityForKubernetesClientAssertion(string? filePath)
+        /// <param name="filePath">Path to a file containing the signed assertion.</param>
+        /// <param name="logger">Logger.</param>
+        public AzureIdentityForKubernetesClientAssertion(string? filePath, ILogger? logger = null)
         {
+            _logger = logger;
+
+            if (filePath == null)
+            {
+                Log.SignedAssertionFileDiskPathNotProvided(_logger);
+            }
+
+            _filePath = _filePath ?? Environment.GetEnvironmentVariable("AZURE_ACCESS_TOKEN_FILE");
```

Comment:
>AZURE_ACCESS_TOKEN_FILE [](http://example.com/codeflow?start=73&length=23)

can these be constants in this class? or a constants file?

Reply:
Done!